README.md

```diff
@@ -1,8 +1,101 @@
 # bn.js [![Build Status](https://secure.travis-ci.org/indutny/bn.js.png)](http://travis-ci.org/indutny/bn.js)
+> BigNum in pure javascript
 
-Just a bike-shed.
+## Install
+`npm install --save bn.js`
 
-#### LICENSE
+## API
+
+```js
+const BN = require('bn.js');
+
+// Numbers
+new BN(12345).toString(16);     // '3039'
+new BN(0x4123456).toString(16); // '4123456'
+
+// Strings
+new BN('FF', 16).toString(); // '255'
+new BN('1A6B765D8CDF', 16).toString(); // '29048849665247'
+
+// Big endian
+new BN([1,2,3,4]).toString(16)); // '1020304'
+new BN([1,2,3,4]).toArray().join(','); // '1,2,3,4'
+
+// Little endian
+new BN([1,2,3], 10, 'le').toString(16); // '30201'
+new BN([1,2,3,4], 10, 'le').toString(16); // '4030201'
+
+// bitLength
+new BN(0x123456).bitLength(); // 21
+new BN('123456789', 16).bitLength(); // 33;
+
+// iaddn
+new BN(-100).sign  // true
+new BN(100).sign   // false
+
+// isubn
+new BN(-100).isubn(200) // -300
+
+// add
+new BN(14).add(new BN(26)).toString(16); // '28'
```

Comment:
Considering that many people will try it out in io.js/node.js repl. May be we could just write?

```javascript
new BN(14).add(new BN(26));  // <BN: 0x28>
```